src/js/components/modal.js

```diff
@@ -82,8 +82,25 @@ var Modal = (function() {
 
     // Sets a class on the body to handle overflow and scroll.
     document.body.classList.add('rvt-modal-open');
-
-    fireCustomEvent(modal.trigger, TRIGGER_ATTR, 'modalOpen');
+    
+    /**
+     * TODO: Here we are checking to see if the modal was triggered by
+     * clicking an element with the "data-modal-trigger" attribute. This
+     * is to cover the case that someone might use a modal without a
+     * trigger associated with it i.e. opening the modal programmatically
+     * via the API. This could probably be refactored to always use the
+     * modal's "id" attribute when emitting the custom event. Same for the
+     * ".close()" method below.
+     */
+    
+    // If there is a corresponding modal trigger
+    if (activeTrigger !== null) {
+      // Fire the custom event from the trigger
+      fireCustomEvent(activeTrigger, TRIGGER_ATTR, 'modalOpen');
```

Comment:
This seems like an odd choice. The modal is the thing that is opening or closing, so the event should only dispatch from the modal, not the trigger. That would simplify what Rivet users have to do. They don't have to think about if they have a trigger or not and where the event is bubbling from. It should be from the one, guaranteed place (the modal).

Reply:
Right, that's why I put that comment explaining it above. I'm hesitant to totally change the functionality now, but I guess there really would never be a time when there wouldn't be an `id` attribute present on the modal itself. I just wouldn't work. It just makes me nervous 😬

Reply:
If there's no `id` on the modal, then throw an error.

Reply:
We're already doing that in the `.open()` method, so I guess it's covered in that sense.

https://github.com/indiana-university/rivet-source/blob/develop/src/js/components/modal.js#L66